library/core/src/marker.rs

```diff
@@ -79,7 +79,7 @@ macro marker_impls {
     on(_Self = "std::rc::Rc<T, A>", note = "use `std::sync::Arc` instead of `std::rc::Rc`"),
     message = "`{Self}` cannot be sent between threads safely",
     label = "`{Self}` cannot be sent between threads safely",
-    note = "consider using `std::sync::Arc<{Self}>`; for more information visit \
+    note = "consider whether `std::sync::Arc<{Self}>` could be incorporated to share this value between threads; for more information visit \
```

Comment:
As long as `Arc<{Self}>` continues to be mentioned, this does not fix #114687. That suggestion is *always* wrong.